### PR TITLE
Bugfix/consul service reg fails tls

### LINKED ISSUE #153: Yorc consul service registration fails if using TLS
# Issue Type

* [x] Bug report
* [ ] Documentation issue report
* [ ] Improvement request

## Description

When Yorc is configured to run in TLS then the registered TCP check in Consul fails

## Expected behavior

Consul service check should be OK

## Actual behavior

check fails

## Steps to reproduce the issue

1. [Configure Yorc to run in TLS](https://yorc.readthedocs.io/en/latest/secured.html#secured-yorc-setup)
2. Start Consul
3. start yorc

In Yorc logs:
```
sept. 03 15:32:40 lal-yorc yorc[9135]: 2018/09/03 15:32:40 http: TLS handshake error from [::1]:54538: EOF
sept. 03 15:32:45 lal-yorc yorc[9135]: 2018/09/03 15:32:45 http: TLS handshake error from [::1]:54540: EOF
sept. 03 15:32:50 lal-yorc yorc[9135]: 2018/09/03 15:32:50 http: TLS handshake error from [::1]:54544: EOF
sept. 03 15:32:55 lal-yorc yorc[9135]: 2018/09/03 15:32:55 http: TLS handshake error from [::1]:54546: EOF
sept. 03 15:33:00 lal-yorc yorc[9135]: 2018/09/03 15:33:00 http: TLS handshake error from [::1]:54548: EOF
```

In Consul logs
```
2018/09/03 15:34:50 [ERR] http: Request PUT /v1/session/create?dc=dc1, error: Check 'service:yorc' is in critical state from=127.0.0.1:44162
2018/09/03 15:34:52 [INFO] agent: Synced check "service:yorc"
```
## Yorc configuration file

```yaml
resources_prefix: lalyorc-
http_port: "8800"
server_id: "lal-yorc"
key_file: /etc/yorc/yorc-server.key
cert_file: /etc/yorc/yorc-server.pem
ca_file: /etc/yorc/ca.pem
ssl_verify: true
consul:
  address: "127.0.0.1:8500"
ansible:
  use_openssh: true

```

### PR DESCRIPTION
**Pull request by adidanes**
# Pull Request description
Fix Issue https://github.com/ystia/yorc/issues/153 

## Description of the change
Update Consul to last available version (1.2.3) to take advantage of this [Consul bug fix.](https://github.com/hashicorp/consul/issues/3364)

Replace TCP checks by HTTP checks for Yorc service

### What I did
- Updated Consul version to 1.2.3
- Created a Health endpoint in the Yorc REST API
- Replaced the TCP check by a HTTP check that calls the Health endpoint
- Updated Yorc documentation

### How to verify it
- Set-up a secured Yorc cluster (secured Consul cluster and secured Yorc instances). If you configure Yorc with ssl_verify set to true, you need to set enable_agent_tls_for_checks to true in the Consul client configuration : 

```json
{
  "domain": "yorc",
  "data_dir": "/tmp/consul",
  "client_addr": "0.0.0.0",
  "advertise_addr": "10.197.138.xxx",
  "retry_join": [ "10.197.138.yyy" ],
  "ports": {
     "https": 8543
  },
  "ca_file": "/etc/yorc/ca.pem",
  "key_file": "/etc/yorc/consul-client.key",
  "cert_file": "/etc/yorc/consul-client.pem",
  "enable_agent_tls_for_checks": true,
  "verify_outgoing": true,
  "verify_incoming_rpc": true,
  "ui": true
}

```

| YORC | http | https | ssl_verify |
|  :---: |  :---: |  :---: |  :---: |
| Consul http | ok | ok | KO |
| Consul https | ok | ok | ok if enable_agent_tls_for_checks |

### Description for the changelog
1. Updated Consul version to 1.2.3
2. Created a Health endpoint in the Yorc REST API
3. Replaced the TCP check by a HTTP check that calls the Health endpoint
4. Update Yorc Docker file to use updated version of Consul in the generated docker image
5. Updated Yorc documentation

## Applicable Issues
 
Fixes #153

### CODE REVIEW

**Comment by loicalbertin** [comment]
> What's your feeling about using a specific constant of PubMaxRoutines only for tests: https://github.com/ystia/yorc/blob/bugfix/consul_service_reg_fails_tls/testutil/helper.go#L29
> 
> With 200 the tests are passing. I didn't test with a bigger value yet...

I feel not confortable by doing this as if it could fail in tests then it will eventually fail one day in production. so we ever have to reduce the default value even for the  main program or investigate why we have such issues.

That makes me think that the issue may come from the way we get the client in tests we do not tweak its configuration as we do in the main prog. I have to check this.

File: doc/secured.rst [hunk 1]
```diff
@@ -22,41 +24,39 @@ You might already have one, otherwise, create it using OpenSSL commands below:
...
     {
-      "domain": "starlings",
+      "domain": "yorc",
       "data_dir": "/tmp/work",
       "client_addr": "0.0.0.0",
       "advertise_addr": "{SERVER_IP}",
       "server": true,
       "bootstrap": true,
-      "encrypt": "{ENCRYPT_KEY}",
+      "ui": true,
```
Comment (loicalbertin) [inline doc/secured.rst]: Why replacing the encrypt key by the ui?
UI is not necessary isn't it?

File: doc/secured.rst [hunk 2]
```diff
@@ -67,18 +67,16 @@ You may find below a typical configuration file for a consul server:
...
 
     {
-      "domain": "starlings",
+      "domain": "yorc",
       "data_dir": "/tmp/work",
       "client_addr": "0.0.0.0",
       "advertise_addr": "{IP}",
-      "ui": true,
       "retry_join": [ "{SERVER_IP}" ],
-      "encrypt": "{ENCRYPT_KEY}",
```
Comment (loicalbertin) [inline doc/secured.rst]: Why not setting up the encryption key for Consul?

**Comment by stebenoist** [comment]
Consul Health checks work well in secured mode. I agree with Loic, we need our tests to stay closer to real-life.

**Pull request merged**